netket/operator/_local_operator/convert.py

```diff
@@ -194,17 +192,9 @@ def local_operators_to_pauli_strings(hilbert, operators, acting_on, constant, dt
         pauli_strings.append("I" * hilbert.size)
         weights.append(constant)
 
-    # the calculation above returns complex coefficients even for operators that are
-    # purely real. So we suppress complexwarnings in that case
+    # the calculation above returns complex weights even for operators that are
+    # purely real, so we discard their imaginary parts
+    weights = [x.real if x.imag == 0 else x for x in weights]
```

Comment:
is it ok to use `.imag==0` as a condition ? 
I think its safe because the imagery part should be identically zero, but please confirm.

Reply:
Let's use `np.isreal` for explicitness. Actually the source of `np.isreal` is just `imag(x) == 0`. If a complex float is made from multiplying a real float with a few 1j, then either its real part or its imaginary part will be exact float zero.